open_humans/views.py

```diff
@@ -283,15 +280,36 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class DataRetrievalTaskDeleteView(PrivateMixin, DeleteView):
+class SourceDataFilesDeleteView(PrivateMixin, DeleteView):
     """
-    Let the user delete a dataset.
+    Let the user delete all datafiles for a source.
     """
+    template_name = 'member/my-member-source-data-files-delete.html'
     success_url = reverse_lazy('my-member-research-data')
 
     def get_queryset(self):
-        return DataRetrievalTask.objects.filter(user=self.request.user)
+        source = self.kwargs['source']
+
+        data_file_model = app_name_to_data_file_model(source)
+
+        return data_file_model.objects.filter(
+            user_data__user=self.request.user)
+
+    def get_object(self, queryset=None):
```

Comment:
Hah, this looks like a hack... but because queryset also has a delete() method I guess it works?

To be safe, I wonder if it's better to do it explicitly rather than trusting DeleteView to always have the right behavior. Not a big deal, maybe just make a comment in the code noting it's a hack (unless I'm wrong about that)